lib/astc_encode.cpp

```diff
@@ -803,3 +884,232 @@ ktxTexture2_CompressAstc(ktxTexture2* This, ktx_uint32_t quality) {
+    if (colorModel != KHR_DF_MODEL_ASTC && This->supercompressionScheme != KTX_SS_NONE) // No supercompression supported yet
+    {
+        return KTX_INVALID_OPERATION; // Not in valid astc decodable format
+    }
+
+    DECLARE_PRIVATE(priv, This);
+
+	uint32_t channelId = KHR_DFDSVAL(BDB, 0, CHANNELID);
+	if (channelId == KHR_DF_CHANNEL_ASTC_DATA) {
+		// Found astc data
+	}
+	else
+		return KTX_FILE_DATA_ERROR;
+
+    // Create a prototype texture to use for calculating sizes in the target
+    // format and, as useful side effects, provide us with a properly sized
+    // data allocation and the DFD for the target format.
+    ktxTextureCreateInfo createInfo;
+    createInfo.glInternalformat = 0;
+    createInfo.vkFormat = vkformat;
+    createInfo.baseWidth = This->baseWidth;
+    createInfo.baseHeight = This->baseHeight;
+    createInfo.baseDepth = This->baseDepth;
+    createInfo.generateMipmaps = This->generateMipmaps;
+    createInfo.isArray = This->isArray;
+    createInfo.numDimensions = This->numDimensions;
+    createInfo.numFaces = This->numFaces;
+    createInfo.numLayers = This->numLayers;
+    createInfo.numLevels = This->numLevels;
+    createInfo.pDfd = nullptr;
+
+    KTX_error_code result;
+    ktxTexture2* prototype;
+    result = ktxTexture2_Create(&createInfo, KTX_TEXTURE_CREATE_ALLOC_STORAGE, &prototype);
+
+    if (result != KTX_SUCCESS) {
+        assert(result == KTX_OUT_OF_MEMORY); // The only run time error
+        return result;
+    }
+
+    if (!This->pData) {
+        if (ktxTexture_isActiveStream((ktxTexture*)This)) {
+             // Load pending. Complete it.
+            result = ktxTexture2_LoadImageData(This, NULL, 0);
+            if (result != KTX_SUCCESS)
+            {
+                ktxTexture2_Destroy(prototype);
+                return result;
+            }
+        } else {
+            // No data to transcode.
+            ktxTexture2_Destroy(prototype);
+            return KTX_INVALID_OPERATION;
+        }
+    }
+
+	// This is where I do the decompression from "This" to prototype target
+    astcenc_profile profile{ASTCENC_PRF_LDR_SRGB};
+    astcenc_swizzle swizzle{ASTCENC_SWZ_R, ASTCENC_SWZ_G, ASTCENC_SWZ_B, ASTCENC_SWZ_A};
+
+    uint32_t        block_size_x{4}; // Get the right blocks from vkformat
+    uint32_t        block_size_y{4};
+    uint32_t        block_size_z{1};
+    float           quality{ASTCENC_PRE_MEDIUM};
+    uint32_t        flags{0}; // TODO: Use normals mode to reconstruct normals params->normalMap ? ASTCENC_FLG_MAP_NORMAL : 0};
+
+	astcBlockDimensions((VkFormat)vkformat, block_size_x, block_size_y, block_size_z);
+
+    // quality = astcQuality(params->qualityLevel);
+    // profile = astcEncoderAction(*params, BDB);
+    // swizzle = astcSwizzle(*params);
+
+    // if(params->perceptual) flags |= ASTCENC_FLG_USE_PERCEPTUAL;
+
+	uint32_t threadCount{1}; // Decompression isn't the bottlneck and only used when checking for psnr and ssim
```

Comment:
This is very annoying. I have https://www.emacswiki.org/emacs/SmartTabs on, as we spoke about it before as well. I still think this is the best way to indent code but it needs to be done for the whole project. We spoke about this before doing such indentation via clang-format. I created https://github.com/KhronosGroup/KTX-Software/pull/545 PR. It was closed without any conlcusion though.

Reply:
Replying just to this for now. Will look over the other changes later.

I closed #545 because there had been many changes to the code you'd opened. Since then Khronos has contracted with RasterGrid to set up something clang-format based to ensure uniform code style. That work has not yet started.

Until then the pre-commit hook seems like a good idea.